pynav/filters.py

```diff
@@ -25,6 +32,41 @@ def check_outlier(error: np.ndarray, covariance: np.ndarray):
     return is_outlier
 
 
+def mean_state(x_array: List[State], weights: np.ndarray) -> np.ndarray:
+    """Computes a weighted mean of an array  of states.
+    It does so in an iterated manner, till the difference is
+    less than a tolerance
+
+    Args:
```

Comment:
very important to use the numpy-style docstring format (see other code), so that our automatic documentation generator can read them properly